becquerel/core/spectrum.py

```diff
@@ -203,6 +203,11 @@ def __init__(self, counts=None, cps=None, uncs=None, bin_edges_kev=None,
         else:
             self.infilename = None
 
+        # These two lines make sure operators between a Spectrum
+        # and a numpy arrays are forbidden and cause a TypeError
+        self.__array_ufunc__ = None
+        self.__array_priority__ = 1
```

Comment:
So this only allows scalar operations?

Reply:
These two attributes only govern the behavior between a custom class and numpy arrays. There is a bit more information [here](https://docs.scipy.org/doc/numpy/reference/arrays.classes.html#numpy.class.__array_ufunc__). 

lf you have something like `anyobj * myobj`, then python checks for the `anyobj.__mul__` method and only if that one returns `NotImplemented` it looks for `myclass.__rmul__`. However, `numpy.arrays()` have a generic `__mul__` method that literally works on any python object, so we have to tell numpy that it should not call that method but raise an error instead. The second line is the method used in older version of numpy, ufuncs are a more modern way.

Limiting multiplication to scalars as far as we can is implemented in the __mul__ method.